src/c_api.cpp

```diff
@@ -1154,19 +1156,21 @@ int LGBM_DatasetPushRowsByCSRWithMetadata(DatasetHandle dataset,
   if (!data) {
     Log::Fatal("data cannot be null.");
   }
-  const int num_omp_threads = OMP_NUM_THREADS();
   auto p_dataset = reinterpret_cast<Dataset*>(dataset);
   auto get_row_fun = RowFunctionFromCSR<int>(indptr, indptr_type, indices, data, data_type, nindptr, nelem);
   int32_t nrow = static_cast<int32_t>(nindptr - 1);
   if (p_dataset->has_raw()) {
     p_dataset->ResizeRaw(p_dataset->num_numeric_features() + nrow);
   }
+
+  const int max_omp_threads = p_dataset->omp_max_threads();
```

Comment:
I think a better fix is to use 
```c++
const int max_omp_threads = p_dataset->omp_max_threads() > 0 ? p_dataset->omp_max_threads(): OMP_NUM_THREADS();
```
And  p_dataset->omp_max_threads()  is initialized to -1, which means use the OMP_NUM_THREADS.
I would not like the change affect the non-distributed version. 
In streaming/distributed version, you can use init_stream to reset the value.

Reply:
ok, I can do that, and we can set the default value on our SynapseML side. Note though, that this IS the streaming method I added, so it is designed specifically to be run with InitStreaming() or it crashes. :)  It just won't work multi-threaded now unless you set a specific value > 0 in InitStreaming(), but it will work fine single-threaded.